### PR TITLE
VM: Implement BOOLAND, BOOLOR opcode

### LINKED ISSUE #226: VM: implement BOOLAND , BOOLOR opcode

### PR DESCRIPTION
**Pull request by dauTT**
Closes #226

### CODE REVIEW

File: pkg/vm/stack/boolean.go [hunk 1]
```diff
@@ -30,3 +30,17 @@ func (b *Boolean) Value() bool {
 func (b *Boolean) Not() *Boolean {
 	return NewBoolean(!b.Value())
 }
+
+// And return a Boolean whose underlying value is obtained
```
Comment (decentralisedkev) [inline pkg/vm/stack/boolean.go]: typo: returns

File: pkg/vm/stack/boolean.go [hunk 2]
```diff
@@ -30,3 +30,17 @@ func (b *Boolean) Value() bool {
...
+// by applying the && operator to two Booleans' values.
+func (b *Boolean) And(a *Boolean) *Boolean {
+	c := b.Value() && a.Value()
+	return NewBoolean(c)
+}
+
+// Or return a Boolean whose underlying value is obtained
```
Comment (decentralisedkev) [inline pkg/vm/stack/boolean.go]: typo: returns

**Comment by decentralisedkev** [comment]
LGTM

File: pkg/vm/stack/boolean.go (hunk 1 above)
Comment (dauTT) [reply pkg/vm/stack/boolean.go]: :+1:

File: pkg/vm/stack/boolean.go (hunk 2 above)
Comment (dauTT) [reply pkg/vm/stack/boolean.go]: :+1:

**Comment by dauTT** [comment]
Ready to be merged.

**Comment by hal0x2328** [comment]
100 points dauTT

**Comment by lllwvlvwlll** [comment]
https://neoscan.io/transaction/630c74684cc6dad330397c63e8c7aa6403c043cc1f241638cff87e3b97a40716

**Pull request merged**